src/ape/api/query.py

```diff
@@ -2,19 +2,46 @@
 
 import pandas as pd
 from ethpm_types.abi import EventABI, MethodABI
-from pydantic import BaseModel, NonNegativeInt, root_validator
+from pydantic import BaseModel, NonNegativeInt, root_validator, validator
 
-from ape._compat import Literal
 from ape.types import AddressType
 from ape.utils import BaseInterfaceModel, abstractmethod
 
+from .providers import BlockAPI
+from .transactions import TransactionAPI
+
 QueryType = Union["BlockQuery", "AccountQuery", "ContractEventQuery", "ContractMethodQuery"]
 
 
 class _BaseQuery(BaseModel):
-    type: str  # Used as discriminator
+
     columns: List[str]
 
+    @classmethod
+    @abstractmethod
+    def all_fields(cls) -> List[str]:
+        """
+        Validates fields that are called during a block query.
+
+        Returns:
+            List[str]: list of columns to be returned in pandas
+                dataframes during block query.
+        """
+        ...
+
+    @validator("columns")
+    def check_columns(cls, data: str) -> Union[List[str], str]:
```

Comment:
Shouldn't be a Union
```suggestion
    def check_columns(cls, data: List[str]) -> List[str]:
```